presto-native-execution/presto_cpp/main/LinuxMemoryChecker.cpp

```diff
@@ -0,0 +1,183 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#include <boost/regex.hpp>
+#include <folly/Conv.h>
+#include <folly/CppAttributes.h>
+#include <folly/File.h>
+#include <folly/FileUtil.h>
+#include <folly/Format.h>
+#include <folly/Range.h>
+#include <folly/String.h>
+#include <folly/gen/Base.h>
+#include <folly/gen/File.h>
+#include <folly/gen/String.h>
+#include "presto_cpp/main/PeriodicMemoryChecker.h"
+#include "presto_cpp/main/common/Configs.h"
+
+#include <sys/stat.h>
+
+namespace facebook::presto {
+
+class LinuxMemoryChecker : public PeriodicMemoryChecker {
+ public:
+  explicit LinuxMemoryChecker(
+      const PeriodicMemoryChecker::Config& config,
```

Comment:
Let's remove the other arguments to the constructor if they are never set.

Reply:
@majetideepak The getters are defined as pure virtual functions in the parent class PeriodicMemoryChecker so that means LinuxMemoryChecker class must define them. I think we should leave the setters as well for this case.

I will remove the other arguments besides `const PeriodicMemoryChecker::Config& config` for LinuxMemoryChecker's constructor since PeriodicMemoryChecker's constructor only has that argument.

Reply:
Let's make the getters virtual functions. It is odd to make them pure virtual if the implementations don't need them.

Reply:
@majetideepak Ok! Implemented the getter virtual functions in PeriodicMemoryChecker base class and removed the other arguments in LinuxMemoryChecker's constructor.